src/correlation_repository.ts

```diff
@@ -168,6 +169,59 @@ export class CorrelationRepository implements ICorrelationRepository, IDisposabl
     await this.correlation.destroy(queryParams);
   }
 
+  public async getCorrelationsByState(state: Runtime.Types.CorrelationState): Promise<Array<Runtime.Types.CorrelationFromRepository>> {
+    const queryParams: Sequelize.FindOptions<ICorrelationAttributes> = {
+      where: {
+        state: state,
+      },
+    };
+
+    const matchingCorrelations: Array<Correlation> = await this.correlation.findAll(queryParams);
+    const correlationsWithState: Array<Runtime.Types.CorrelationFromRepository> =
+      matchingCorrelations.map(this._convertTocorrelationRuntimeObject.bind(this));
+
+    return correlationsWithState;
+  }
+
+  public async finishCorrelation(correlationId: string): Promise<void> {
+    const queryParams: Sequelize.FindOptions<ICorrelationAttributes> = {
+      where: {
+        correlationId: correlationId,
+      },
+    };
+
+    const correlationWithId: Correlation = await this.correlation.findOne(queryParams);
+
+    const noMatchingCorrelationFound: boolean = correlationWithId === undefined;
+    if (noMatchingCorrelationFound) {
+      throw new NotFoundError(`No matching correlation with ID ${correlationId}`);
```

Comment:
```suggestion
      throw new NotFoundError(`No matching correlation with ID ${correlationId} found!`);
```

Full sentences are cool. 😉